File: C++/프로그래머스/0/120862. 최댓값 만들기 （2）/최댓값 만들기 （2）.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

int solution(vector<int> numbers) {
    sort(numbers.begin(), numbers.end());
    
    int end_idx = numbers.size() - 1;
    
    int min_s = numbers[0] * numbers[1];
    int max_s = numbers[end_idx] * numbers[end_idx - 1];
    
    return max(max_s, min_s);
}

File: Go/프로그래머스/0/120862. 최댓값 만들기 （2）/최댓값 만들기 （2）.go
import "sort"

func max(a, b int) int {
    if a > b {
        return a
    }
    return b
}

func solution(numbers []int) int {
    // 원본 슬라이스 훼손 방지
    res := make([]int, len(numbers))
    copy(res, numbers)
    
    sort.Ints(res)
    
    l := len(res)
    
    a := res[0] * res[1]
    b := res[l - 1] * res[l - 2]
    
    return max(a, b)
}